aiida/engine/processes/workchains/workchain.py

```diff
@@ -121,24 +121,53 @@ def on_run(self):
         super().on_run()
         self.node.set_stepper_state_info(str(self._stepper))
 
+    def _resolve_nested_context(self, key: str) -> Tuple[AttributeDict, str]:
+        """
+        Returns a reference to a sub-dictionary of the context and the last key,
+        after resolving a potentially segmented key where required sub-dictionaries are created as needed.
+
+        :param key: A key into the context, where words before a dot are interpreted as a key for a sub-dictionary
+        """
+        ctx = self.ctx
+        ctx_path = key.split('.')
+
+        for path in ctx_path[:-1]:
+            try:
+                ctx = ctx[path]
+            except TypeError:
+                raise AssertionError(
```

Comment:
Shouldn't this really be a `ValueError`? This is raised when the `key` is inconsistent with requested action and `key` is a user supplied value from within the `WorkChain` definition.

Reply:
Agreed, I was looking at the surround/previous code where `assert ...` were used so I followed that guideline. A `KeyError` would be the other logical choice.

Reply:
done.